src/Models/HydrostaticFreeSurfaceModels/distributed_split_explicit_free_surface.jl

```diff
@@ -0,0 +1,87 @@
+using Oceananigans.AbstractOperations: GridMetricOperation, Δz
+using Oceananigans.Distributed: DistributedGrid
+using Oceananigans.Models.HydrostaticFreeSurfaceModels: SplitExplicitState, SplitExplicitFreeSurface
+
+import Oceananigans.Models.HydrostaticFreeSurfaceModels: FreeSurface, SplitExplicitAuxiliary
+
+function SplitExplicitAuxiliary(grid::DistributedGrid)
+    
+    Gᵁ = Field((Face,   Center, Nothing), grid)
+    Gⱽ = Field((Center, Face,   Nothing), grid)
+    
+    Hᶠᶜ = Field((Face,   Center, Nothing), grid)
+    Hᶜᶠ = Field((Center, Face,   Nothing), grid)
+    Hᶜᶜ = Field((Center, Center, Nothing), grid)
+    
+    vertical_height!(Hᶠᶜ, (Face, Center, Center))
+    vertical_height!(Hᶜᶠ, (Center, Face, Center))
+
+    vertical_height!(Hᶜᶜ, (Center, Center, Center))
```

Comment:
Are there other heights? Horizontal height?

Reply:
no just that

Reply:
Should we use a different name? `calculate_column_height` or something?

Reply:
@navidcy suggests:

```julia
calculate_column_height!(Hᶜᶠ, (Center, Face, Center), ...)
```

?

We are also confused: is `Hᶜᶠ` a field (so this uses `Hᶜᶠ.grid`?) Why do we also need to pass the location?

Finally I'll note that `Hᶜᶠ` is ambiguous (unless you understand the code in detail) because it is missing the third location. So someone glancing at the code cannot interpret this without the third letter to disambiguate (or reading the code in detail to understand). Here would be `a` for the vertical location.

Reply:
This function integrates `z` into `H`. `Hᶜᶠ` and so on are defined in `split_explicit_free_surface.jl`. It is a field because it has a location. The location is passed because you want to integrate the correct `Δz` corresponding to the depth location. It is a 2D field (reduced in the vertical) so `a` might be misleading.

Reply:
Is there anywhere else in the code we have a reduced field with `cf`, etc subscripts?